pkg/debuggers/python/impl.go

```diff
@@ -0,0 +1,96 @@
+package python
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/solo-io/squash/pkg/debuggers"
+)
+
+const (
+	PtvsdPortEnvVariable = "PTVSD_PORT_NUMBER"
+	PtvsdSearchString    = `ptvsd\.enable_attach.*`
+)
+
+type PythonInterface struct{}
+
+type ptvsdDebugServer struct {
+	port int
+}
+
+func (d *ptvsdDebugServer) Detach() error {
+	return nil
+}
+
+func (d *ptvsdDebugServer) Port() int {
+	return d.port
+}
+
+func (d *ptvsdDebugServer) HostType() debuggers.DebugHostType {
+	return debuggers.DebugHostTypeTarget
+}
+
+func (i *PythonInterface) Attach(pid int) (debuggers.DebugServer, error) {
+
+	log.WithField("pid", pid).Debug("AttachToLiveSession called")
+	port, err := getPtvsdPort(pid)
+	if err != nil {
+		log.WithField("err", err).Error("can't get ptvsd port")
+		return nil, err
+	}
+
+	log.WithFields(log.Fields{"pid": pid, "port": port}).Debug("Found python debug port")
+	ds := &ptvsdDebugServer{
+		port: port,
+	}
+	return ds, nil
+}
+
+// Search /proc/{PID}/cwd for file with "ptvsd.enable_attach" string and extracts port from it
+func getPtvsdPort(pid int) (int, error) {
+	port := 0
+
+	// Try environment var first
+	pe := os.Getenv(PtvsdPortEnvVariable)
+	if len(pe) >= 1 {
+		fmt.Sscanf(pe, "%d", &port)
+		return port, nil
+	}
+
+	root := filepath.Join("/proc", fmt.Sprintf("%d", pid), "cwd")
+	log.WithField("root", root).Debug("searching root")
+	re := regexp.MustCompile(PtvsdSearchString)
+
+	filepath.Walk(root+"/", func(p string, fi os.FileInfo, err error) error {
+		if err != nil || fi == nil || fi.IsDir() {
```

Comment:
consider checking just for *py files?

Reply:
I thought about it and I'll probably do it, but than: http://dcjtech.info/topic/python-file-extensions/
I can also try to ignore known compiled files, like pyc, pyo etc. instead.